src/transformers/models/superpoint/modeling_superpoint.py

```diff
@@ -0,0 +1,510 @@
+    scores, indices = torch.topk(scores, k, dim=0)
+    return keypoints[indices], scores
+
+
+def simple_nms(scores: torch.Tensor, nms_radius: int) -> torch.Tensor:
+    """Applies non-maximum suppression on scores"""
+    if nms_radius < 0:
+        raise ValueError("Expected positive values for nms_radius")
+
+    def max_pool(x):
+        return nn.functional.max_pool2d(x, kernel_size=nms_radius * 2 + 1, stride=1, padding=nms_radius)
+
+    zeros = torch.zeros_like(scores)
+    max_mask = scores == max_pool(scores)
+    for _ in range(2):
+        supp_mask = max_pool(max_mask.float()) > 0
+        supp_scores = torch.where(supp_mask, zeros, scores)
+        new_max_mask = supp_scores == max_pool(supp_scores)
+        max_mask = max_mask | (new_max_mask & (~supp_mask))
+    return torch.where(max_mask, scores, zeros)
+
+
+@dataclass
+class ImagePointDescriptionOutput(ModelOutput):
+    """
+    Base class for outputs of image point description models. Due to the nature of keypoint detection, the number of
+    keypoints is not fixed and can vary from image to image, which makes batching non-trivial. In the batch of images,
+    the maximum number of keypoints is set as the dimension of the keypoints, scores and descriptors tensors. The mask
+    tensor is used to indicate which values in the keypoints, scores and descriptors tensors are keypoint information
+    and which are padding.
+
+    Args:
+        keypoints (`torch.FloatTensor` of shape `(batch_size, num_keypoints, 2)`):
+            Relative (x, y) coordinates of predicted keypoints in a given image.
+        scores (`torch.FloatTensor` of shape `(batch_size, num_keypoints)`):
+            Scores of predicted keypoints.
+        descriptors (`torch.FloatTensor` of shape `(batch_size, num_keypoints, descriptor_size)`):
+            Descriptors of predicted keypoints.
+        mask (`torch.BoolTensor` of shape `(batch_size, num_keypoints)`):
+            Mask indicating which values in keypoints, scores and descriptors are keypoint information.
+        last_hidden_state (`torch.FloatTensor` of shape `(batch_size, sequence_length, hidden_size)`):
+            Sequence of hidden-states at the output of the last layer of the decoder of the model.
+        hidden_states (`tuple(torch.FloatTensor)`, *optional*, returned when `output_hidden_states=True` is passed or
+        when `config.output_hidden_states=True`):
+            Tuple of `torch.FloatTensor` (one for the output of the embeddings, if the model has an embedding layer, +
+            one for the output of each stage) of shape `(batch_size, sequence_length, hidden_size)`. Hidden-states
+            (also called feature maps) of the model at the output of each stage.
+    """
+
+    keypoints: Optional[torch.IntTensor] = None
+    scores: Optional[torch.FloatTensor] = None
+    descriptors: Optional[torch.FloatTensor] = None
+    mask: Optional[torch.BoolTensor] = None
+    last_hidden_state: torch.FloatTensor = None
```

Comment:
last_hidden_states should be the first returned value here

Reply:
Fixed in [this commit](https://github.com/huggingface/transformers/pull/28966/commits/c8a21216ac36870e0e87537a79de46593bb6f906)